packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -3227,6 +3231,413 @@ void main() {
           ),
         ),
       );
+
+      tester.binding.window.physicalSizeTestValue = null;
+      tester.binding.window.devicePixelRatioTestValue = null;
+    });
+  });
+
+  group('textAlignVertical position', () {
+    group('simple case', () {
+      testWidgets('align top (default)', (WidgetTester tester) async {
+        final FocusNode focusNode = FocusNode();
+        const Size size = Size(200.0, 200.0);
+        await tester.pumpWidget(
+          CupertinoApp(
+            debugShowCheckedModeBanner: false,
+            home: CupertinoPageScaffold(
+              child: Align(
+                alignment: Alignment.center,
+                child: SizedBox(
+                  width: size.width,
+                  height: size.height,
+                  child: CupertinoTextField(
+                    focusNode: focusNode,
+                    expands: true,
+                    maxLines: null,
+                  ),
+                ),
+              ),
+            ),
+          ),
+        );
+
+        // Fills the whole container since expands is true.
+        expect(tester.getSize(find.byType(CupertinoTextField)), size);
+
+        // Tapping anywhere inside focuses it.
+        expect(focusNode.hasFocus, false);
+        await tester.tapAt(tester.getTopLeft(find.byType(CupertinoTextField)));
+        await tester.pumpAndSettle(const Duration(milliseconds: 400)); // timer error if no duration
```

Comment:
If I didn't add this Duration, then I got failures due to a timer still existing after the test ended.  Any other way to fix that? Am I doing something weird here?

Reply:
This might be the 300ms doubleTap timer from packages/flutter/lib/src/widgets/text_selection.dart. Pump and settle only waits for 100ms, so the timer may survive to the end of the test. Assuming that's correct, you could explicitly add a second 300 ms pump (after a no-args pumpAndSettle) to ensure that the double tap timer has expired.

Reply:
That seems to be it, thanks for the tip.